legend-engine-core/legend-engine-core-executionPlan-execution/legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/nodes/helpers/freemarker/FreeMarkerExecutor.java

```diff
@@ -71,15 +71,15 @@ else if (v instanceof StreamingResult)
 
     public static String processRecursively(String input, Map<String, ?> variableMap, String templateFunctions)
     {
-        String result = process(input, variableMap, templateFunctions);
+        String result = (templateFunctions == null || templateFunctions == "") && variableMap.isEmpty() ? input : process(input, variableMap, templateFunctions);
```

Comment:
can we move this in above function, here it would have to be processed multiple times?